haystack_experimental/components/extractors/llm_metadata_extractor.py

```diff
@@ -240,28 +245,73 @@ def from_dict(cls, data: Dict[str, Any]) -> "LLMMetadataExtractor":
+
 
     @component.output_types(documents=List[Document], errors=Dict[str, Any])
-    def run(self, documents: List[Document]) -> Dict[str, Any]:
+    def run(self, documents: List[Document], page_range: Optional[List[str]] = None):    # pylint: disable=R0914
         """
         Extract metadata from documents using a Language Model.
 
+        If `page_range` is provided, the metadata will be extracted from the specified range of pages. This component
+        will split the documents into pages and extract metadata from the specified range of pages. The metadata will be
+        extracted from the entire document if `page_range` is not provided.
+
+        The original documents will be returned  updated with the extracted metadata.
+
         :param documents: List of documents to extract metadata from.
+        :param page_range: A range of pages to extract metadata from. For example, page_range=['1', '3'] will extract
+                           metadata from the first and third pages of each document. It also accepts printable range
+                           strings, e.g.: ['1-3', '5', '8', '10-12'] will extract metadata from pages 1, 2, 3, 5, 8, 10,
+                           11, 12.
+                           If None, metadata will be extracted from the entire document for each document in the
+                           documents list.
         :returns:
             A dictionary with the keys:
-            - "documents": List of documents with extracted metadata.
+            - "documents": The original list of documents updated with the extracted metadata.
             - "errors": A dictionary with document IDs as keys and error messages as values.
         """
+
         errors = {}
+
         for document in documents:
-            prompt_with_doc = self.builder.run(input_text=document.content)
-            result = self.llm_provider.run(prompt=prompt_with_doc["prompt"])
-            llm_answer = result["replies"][0]
-            if self.is_valid_json_and_has_expected_keys(expected=self.expected_keys, received=llm_answer):
-                extracted_metadata = json.loads(llm_answer)
-                for k in self.expected_keys:
-                    document.meta[k] = extracted_metadata[k]
+
+            if not document.content:
+                logger.warning(f"Document {document.id} has no content. Skipping metadata extraction.")
+                continue
+
+            if page_range:  # extract metadata from a specific range of pages
+                expanded_page_range = expand_page_range(page_range) if page_range else None
+                start_document = expanded_page_range[0] - 1
+                end_document = expanded_page_range[-1]
+                splitter = DocumentSplitter(split_by="page", split_length=1)
+                pages = splitter.run(documents=[document])
+                merged_contents = [page.content + '\n' for page in pages['documents'][start_document:end_document]]
```

Comment:
This doesn't respect that the page ranges specified could be non-continuous. For example if I specify `page_range=['2', '5']` this code will send pages 2, 3, 4 and 5 and not just 2 and 5.